dashboard/app/models/sections/section.rb

```diff
@@ -109,11 +109,24 @@ def user_must_be_teacher
   end
   validate :user_must_be_teacher, unless: -> {deleted?}
 
-  before_create :assign_code
-  def assign_code
+  before_create :scrub_section
+  def scrub_section
+    # Set defaults if none supplied
+    self.stage_extras = false if stage_extras.nil?
+    self.pairing_allowed = true if pairing_allowed.nil?
```

Comment:
This seems a little confusing to me. Are these meant to be default initial values? If so, set them declaratively in [after_initialize](http://guides.rubyonrails.org/active_record_callbacks.html#after-initialize-and-after-find).

Or are they actually meant to override explicit nil values provided in initializer params? That seems weird to me. Again I'd prefer that to just be invalid and fix the caller.